pkg/processor/build/builder.go

```diff
@@ -307,10 +307,17 @@ func (b *Builder) resolveFunctionPath(functionPath string) (string, error) {
 			"url", functionPath,
 			"target", tempFileName)
 
-		if err := common.DownloadFile(functionPath, tempFileName); err != nil {
+		if err = common.DownloadFile(functionPath, tempFileName); err != nil {
 			return "", err
 		}
 
+		if util.IsCompressed(tempFileName) {
```

Comment:
This seems like duplication that can be avoided. If this block would set `functionPath` to be `tempFileName` after downloading, instead of returning `tempFileName, nil`, it would just be extracted below if needed.